src/unsafe_self_cell.rs

```diff
@@ -170,9 +165,7 @@ impl<Owner, Dependent> Drop for OwnerAndCellDropGuard<Owner, Dependent> {
 
         // Deallocate even when the drop_in_place(...owner) panics
         let _guard = DeallocGuard {
-            ptr: unsafe {
-                transmute::<*mut JoinedCell<Owner, Dependent>, *mut u8>(self.joined_ptr.as_ptr())
-            },
+            ptr: self.joined_ptr.as_ptr().cast(),
```

Comment:
Please specify the cast result type here "ptr: self.joined_ptr.as_ptr().cast(),"